pandas/core/indexes/datetimelike.py

```diff
@@ -239,9 +239,8 @@ def min(self, axis=None, skipna=True, *args, **kwargs):
         i8 = self.asi8
         try:
             # quick check
-            if len(i8) and self.is_monotonic:
-                if i8[0] != iNaT:
-                    return self._data._box_func(i8[0])
+            if len(i8) and self.is_monotonic and i8[0] != iNaT:
+                return self._data._box_func(i8[0])
```

Comment:
wouldnt surprise me if this can go outside the "try", also could check len(self) instead of len(i8)